src/controller/AbstractDnssdDiscoveryController.cpp

```diff
@@ -22,9 +22,34 @@
 #include <lib/core/CHIPEncoding.h>
 #include <lib/support/logging/CHIPLogging.h>
 
+#include <bitset>
+
 namespace chip {
 namespace Controller {
 
+static bool SameExceptOrder(const chip::Span<const Inet::IPAddress> & source, const chip::Span<const Inet::IPAddress> & destination)
+{
+    std::bitset<chip::Dnssd::CommonResolutionData::kMaxIPAddresses> addressUsed;
+    if (source.size() != destination.size())
```

Comment:
please add a verify that `size() <= kMaxIpAddresses` since otherwise the indexing below may go out of bounds.

You could VerifyOrDie or return false if the sizes are out of bounds.

Reply:
I added. Thanks.